### DISCUSSION TITLE
Using the HA REST API with token authorization

### DISCUSSION THREAD

**Discussion by andarotajo** [hassio-addons/app-appdaemon#326] [Q&A] [answered]
Hey, I'm currently writing an app where I use the [HA REST API](https://developers.home-assistant.io/docs/api/rest) to check the configuration. This request requires a token and while I could just add a long-lived token to the `appdaemon.yaml`, the add-on documentation does not recommend overwriting this.

My question now is: Can I do the API request without supplying my own token?

**Comment by frenck** [comment, marked as answer]
The token exists, it is injected by the add-on.

../Frenck

**Reply by andarotajo** [reply]
Thanks for the quick answer. I'm probably doing something wrong but the API call is not working for me.

```python
api_url = "http://127.0.0.1:8123/api/config/core/check_config"
headers = {"content-type": "application/json"}
response = requests.post(api_url, headers=headers)
```

**Reply by frenck** [reply]
That is not posting to the internal API, you are going outside of it.

The URL you should be using is `http://supervisor/core/api`

**Reply by andarotajo** [reply]
Ah didn't know about the internal API. Tried the `/core/check` endpoint but that one didn't work unfortunately. Got it working like this though

```python
token = os.getenv("SUPERVISOR_TOKEN")
api_url = "http://supervisor/core/api/config/core/check_config"
headers = {"Authorization": f"Bearer {token}", "content-type": "application/json"}

response = requests.post(api_url, headers=headers)
```

**Reply by andarotajo** [reply]
Just to make sure that I didn't do anything wrong in my test but endpoints like `/core/check` are just not usable for an AD app right?

I've tried other ways to check the config like the service which doesn't return any result but this seems to be the "best" way in terms of accessibility.

**Reply by frenck** [reply]
Honestly, in general, there, there isn't much good reason to use check config at all...

**Reply by andarotajo** [reply]
I could probably remove it since I use your GitHub action to check the config before pulling the changes into my HA instance but I guess it doesn't hurt keeping this check in.

The idea was to skim over the Git changes and see, if I really need a reboot. I'm using the Git Pull add-on to get the changes but the reboot feature it has just does so for every little thing instead of just reloading something like automations. Haven't found a good automated solution to that yet
